iocore/hostdb/I_HostDBProcessor.h

```diff
@@ -146,16 +146,17 @@ struct HostDBInfo : public RefCountObj {
     ink_release_assert(iobuffer_index >= 0);
     void *ptr = ioBufAllocator[iobuffer_index].alloc_void();
     memset(ptr, 0, size);
-    HostDBInfo *ret     = new (ptr) HostDBInfo();
-    ret->iobuffer_index = iobuffer_index;
+    HostDBInfo *ret      = new (ptr) HostDBInfo();
+    ret->_iobuffer_index = iobuffer_index;
     return ret;
   }
 
   void
   free() override
   {
-    Debug("hostdb", "freeing %d bytes at [%p]", (1 << (7 + iobuffer_index)), this);
-    ioBufAllocator[iobuffer_index].free_void((void *)(this));
+    ink_release_assert(from_alloc());
```

Comment:
Looks like, the diff introduces new ink_release_assert to make sure the object is allocated from HostDBInfo::alloc().
Could we simply the object uncopyable to achieve the same?

Reply:
>The current code requires that instances of HostDBInfo be copyable.
Could you elaborate on that? I'll be surprised to see if any HostDBInfo is copied around. ^_^

Reply:
Here it the case I am aware of where a copy occurs:  https://github.com/apache/trafficserver/blob/af9c4f39aeb40515bb42469cb0aeeb0ac27fec54/proxy/http/HttpSM.cc#L2254